hooks/boot/06-postgresql-upgrade-extensions.rb

```diff
@@ -2,18 +2,23 @@
 
 module PostgresqlUpgradeHookContextExtension
   def needs_postgresql_upgrade?(new_version)
-    File.read('/var/lib/pgsql/data/PG_VERSION').chomp.to_i < new_version.to_i
+    current_version.to_i < new_version.to_i
   rescue Errno::ENOENT
     false
   end
 
+  def current_version
+    File.read('/var/lib/pgsql/data/PG_VERSION').chomp
+  end
+
   def os_needs_postgresql_upgrade?
     el8? && needs_postgresql_upgrade?(13)
   end
 
   def postgresql_upgrade(new_version)
     logger.notice("Performing upgrade of PostgreSQL to #{new_version}")
 
+    execute!("dnf module switch-to postgresql:#{current_version} -y", false, true)
```

Comment:
Do we need to move the PG13 upgrade into f-maintain and have it in two spots based on entry point?

Reply:
No, because the user (w/o maintain) also needs to perform `switch-to 13` to get the new installer.

We need to get rid of this monstrosity called modularity

Reply:
I guess you have figured it all out by now so might as well take this in. My second thought when reading this is -- should we add a check to catch this before the upgrade is started and have the user take a corrective action first?

Reply:
Also, is this user / cluster locale mismatch a known issue within PG land?

Reply:
> I guess you have figured it all out by now so might as well take this in. My second thought when reading this is -- should we add a check to catch this before the upgrade is started and have the user take a corrective action first?

What kind of corrective action? And what check? The system is "broken" the moment switch-to is executed, regardless what executes kt

Reply:
Can we use `systemd-run postgresql --property` to run it temporarily with a different `ExecStart`?

Reply:
> Also, is this user / cluster locale mismatch a known issue within PG land?

Yes. The Debian tooling does this (extracting the locale of the old cluster and using it for the new) automatically, RHEL's does not.

Reply:
And I've opened https://issues.redhat.com/browse/RHEL-58410 to track this problem as a whole

Reply:
Well, turned out starting a cluster in single-user mode is not *that* hard, and solves stupid problems with dnf…

https://github.com/theforeman/foreman-installer/pull/982